packages/yoastseo/src/languageProcessing/helpers/match/matchKeywordWithSentence.js

```diff
@@ -0,0 +1,72 @@
+import { cloneDeep } from "lodash-es";
+
+const wordCouplers = [ "_", "-" ];
+
+/**
+ * Matches a keyword with a sentence object from the html parser.
+ *
+ * @param {(string[])[]} keywordForms The keyword forms.
+ * E.g. If the keyphrase is "key word", then (if premium is activated) this will be [ [ "key", "keys" ], [ "word", "words" ] ]
+ * The forms are retrieved higher up (a.o. in keywordCount.js) with researcher.getResearch( "morphology" ).
+ * @param {Sentence} sentence The sentence to match against the keywordForms.
+ *
+ * @returns {Token[]} The tokens that match the keywordForms.
+ *
+ * The algorithm is as follows:
+ *
+ * It iterates over all tokens in the sentence. It compares the current token with the keyword forms.
+ * If it matches, it adds the token to the matches array.
+ *
+ * The keyword forms are tokenized differenty than the sentence.
+ * The keyword forms are tokenized with researcher.getResearch( "morphology" ) and the sentence is tokenized with the html parser.
+ * This leads to differences in tokenization. For example, the html parser tokenizes "key-word" as [ "key", "-", "word" ]. The morphology
+ * tokenizes it as [ "key-word" ].
+ * This function corrects for these differences by combining tokens that are separated by a word coupler (e.g. "-") into one token: the matchToken.
```

Comment:
Would it be possible to instead align how the tokenizer and the getWords helper tokenize strings? Or is there a reason why they should be different? If there is a reason, maybe you could explain it here?

Reply:
yes, that's a good idea!